packages/openapi3/test/tsp-openapi3/specs/status-code-changes/service.yml

```diff
@@ -0,0 +1,30 @@
+openapi: 3.0.0
+info:
+  title: (title)
+  version: 0.0.0
+tags: []
+paths:
+  /:
+    get:
+      operationId: extensive
+      parameters: []
+      responses:
+        "200":
+          description: The request has succeeded.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Pet"
+        4XX:
```

Comment:
Is this a valid spec?

Could we add some manually written spec for tests? I think the generated specs are:
1. might have some errors
2. it is overfitting the emitter logic, so I'm really concerned that it is not reflecting the actual specs we are expected to target.

But I think this could be improved later, not have to be in this PR.

Reply:
How about convert openapi3 spec to typespec, then convert back to openapi3 spec, and then compares the 2 versions of openapi 3 spec? so that we don't need to worry about the test correctness.

Reply:
> Is this a valid spec?

Yes! OpenAPI3 does allow you to specify _some_ status code ranges:
> To define a range of response codes, this field MAY contain the uppercase wildcard character X. For example, 2XX represents all response codes between [200-299]. Only the following range definitions are allowed: 1XX, 2XX, 3XX, 4XX, and 5XX.

For what it's worth - all of the swagger specs, except the `petstore-sample` which I found online as a complex spec - were generated from TypeSpec files originally.

Reply:
> OpenAPI3 does allow you to specify some status code ranges

Thanks! I didn't know that before.

> all of the swagger specs were generated from TypeSpec files originally.

Yeah that's my concern. For example, you have made up some new models as we discussed above, will you say the generated tsp a good candidate test for tsp-to-open3.0 emitter? Same to the tsp-to-open3.0 emitter, which might have similar magic I don't know, the generated open3.0 spec might not a good candidate test for open3.0-to-tsp emitter.

Reply:
Some additional context - this is meant to be a tool for 3rd parties outside of Azure to quickly migrate their existing OpenAPI3 specs to TypeSpec. We expect that these authors will want to massage their TypeSpec files once they are generated, but want to give them a reasonable starting point. This is not meant to be the tool that converts Azure swaggers into TypeSpec.

With that in mind - I'm trying to make sure that we can convert valid OpenAPI3 into TypeSpec without breaking. The `petstore-swagger` sample was one I found out in the wild as one sanity check against just TypeSpec-generated OpenAPI files.

The TypeSpec generated OpenAPI files do follow some conventions - such as prefixing operationIds with the interface name and an underscore - that won't be applicable to the specs we're expecting to use this tool. We might find that it makes more sense to use tags to create interfaces in this case - but it's not a goal to be able to go TypeSpec -> OpenAPI3 -> TypeSpec and end up with the same file as the original.